### ISSUE TITLE
[Bug] Achievements with an API call as the only condition always show 0%

### ISSUE THREAD

**Issue by UsernamesAreNotMyThing** [Ayagikei/LifeUp#103]
**Describe the bug**
Achievements that have an API call as the only condition always appear as 0% in the list, regardless of actual progress.

**Expected behavior**
The displayed percentage should reflect the progress on the API call condition.

**Screenshots**
![Screenshot_20231016-125959](https://github.com/Ayagikei/LifeUp/assets/92757918/3d451d61-d749-41c7-864d-9fb9b4ff1f47)

**Environments and Version (please complete the following information):**
 - OS: **Android 10**
 - Version **1.93** (was occurring before this update as well)

**Comment by Ayagikei** [comment]
Thanks for the feedback!

We will look into it asap.

**Comment by Ayagikei** [comment]
Hello, I tried to use the item to test the API, but could not reproduce the problem.

We need to confirm, are you using an external tool such as Tasker to call the API?

v1.93.0-rc02 We checked the relevant code and found that there is logic at the code level to update the overall achievement progress. But it and updating the progress of a single condition belong to two processes and are executed by a dedicated single thread. Then fixed an issue that could cause progress calculation errors.

We're not sure yet whether this is caused by single-thread blocking or something else.

**Comment by UsernamesAreNotMyThing** [comment]
Yes, this problem is specifically with the External API Call condition when "running" it from automation apps like Tasker.
